sui_programmability/examples/games_with_chance/sources/satoshi_flip.move

```diff
@@ -0,0 +1,300 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+
+/// Satoshi flip is a fair and secure way to conduct a coin flip.
+/// A player called "house" chooses a random secret and uploads its hash on chain
+/// Another player called "player",  makes a guess on a predetermined bit (eg. last) of the secret
+/// House reveals the secret and the winner is determined
+/// 
+/// This implementation checks the last bit of the first byte of the secret
+module games_with_chance::satoshi_flip {
+    // imports
+    use std::option::{Self, Option};
+    use std::hash::sha3_256;
+    use std::vector;
+
+    use sui::coin::{Self, Coin};
+    use sui::balance::{Self, Balance};
+    use sui::sui::SUI;
+    use sui::digest::{Self, Sha3256Digest};
+    use sui::object::{Self, UID};
+    use sui::tx_context::{Self, TxContext};
+    use sui::transfer;
+    
+    /* Terminology
+        house: The user who picks the secret
+        player: The user who wages an amount on his guess
+        max_bet: the maximum amount the house is willing to potentialy lose or win
+        min_bet: the minimum amount the house is willing to accept as a wager/stake (too low of an amount might be worth the gas) min_bet <= max_bet
+        stake: the amount the player is willing to wager on his guess, min_bet <= stake <= max_bet
+    */
+
+    /*
+        Conventions:
+        A guess = 2 in the outcome means player has not bet but the game was ended
+        A guess = 3 in the outcome means player has bet but house refused to reveal in time
+    */
+
+    /// Hash function the house will use
+    const HASH: vector<u8> = b"SHA3-256";
+
+    /// How many epochs must pass after `fun bet` was called, until the player may cancel a game
+    // TODO: This should be possible to be set by house up to a max limit
+    const EpochsCancelAfter: u64 = 7;
```

Comment:
why 7? is 2 epochs enough? the user may send its bet just before the end of an epoch, and so we can't use here 1, but more than 1 should be safe

Reply:
This was requested by @kchalkias, it is like a standard in the gaming industry.

Reply:
The actual reason is that in practice sometimes a random failure may happen, if it is the weekend and no person is on call the house might lose a winning game. One step further, along with the weekend, holidays might interimpose so the industry in general settled at 7 days. For our demo purposes it makes little difference, so we just went with the standard.

Reply:
> why 7? is 2 epochs enough? the user may send its bet just before the end of an epoch, and so we can't use here 1, but more than 1 should be safe

7 days is the norm for fraud proof systems. We have to take into network outages, system/custody malfunction, lack of funds (so you need to find some to pay for gas), weekends and holidays (where services might be sub operating), potential access to cold keys (which requires some time due to custody deps). @Eis-D-Z please add a comment with the above and I believe we're good.